File: harness/semgrep/fixtures/compare-catch/verify.bad.py
# BAD: non-constant-time secret compare, and except-as-accept (fails open).

import hmac
import hashlib


def check_mac(payload: bytes, secret_key: bytes, provided_mac: str) -> bool:
    expected_mac = hmac.new(secret_key, payload, hashlib.sha256).hexdigest()
    return expected_mac == provided_mac


def verify_token(token: str) -> bool:
    try:
        assert_signature_valid(token)
        return True
    except Exception:
        return True

File: harness/semgrep/fixtures/compare-catch/verify.bad.go
package verify

import (
	"bytes"
	"crypto/hmac"
	"crypto/sha256"
)

// BAD: non-constant-time secret compare, and err-swallowed-as-accept.

func checkMac(payload, secretKey []byte, providedMac []byte) bool {
	mac := hmac.New(sha256.New, secretKey)
	mac.Write(payload)
	expectedMac := mac.Sum(nil)
	// timing-variable
	return bytes.Equal(expectedMac, providedMac)
}

func verifyToken(token string) (bool, error) {
	err := assertSignatureValid(token)
	if err != nil {
		// swallow verification failure as success => fails open
		return true, nil
	}
	return true, nil
}
